spec/lib/middleware_notification_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+require 'middleware_notification'
+
+describe ManageIQ::Providers::Hawkular::MiddlewareNotification do
+  describe ManageIQ::Providers::Hawkular::MiddlewareNotification::NotificationData do
+    subject { described_class.new(:mw_op_success, :Suspend, nil, server) }
+    let(:server) { instance_double('MiddlewareServer') }
+
+    it 'has a server name' do
+      allow(server).to receive(:name) { 'name' }
+      allow(server).to receive(:feed) { 'feed' }
+
+      expect(subject.server_name).to eq 'name (feed)'
+    end
+  end
+
+  describe '.build' do
+    context 'class is MiddlewareServer' do
+    end
+
+    context 'entity found'
+    context 'entity not found'
+  end
+
+  describe '.emit_success'
+  describe '.emit_failure'
+  describe '#emit'
```

Comment:
In standard method notation for Ruby, `Class#something` denotes an instance method, while `Class.something` is a class method or a singleton method.